### PR TITLE
Adding registerExtraComponents to allow registering additional components in various services

### PR DESCRIPTION
**Pull request by soumitra-st**
Adding registerExtraComponents after creating instance of the application and before starting to allow registering extra components to various Pinot services.

`feature`

### CODE REVIEW

File: pinot-minion/src/main/java/org/apache/pinot/minion/BaseMinionStarter.java [hunk 1]
```diff
@@ -281,6 +281,7 @@ public void start()
     minionContext.setHelixManager(_helixManager);
     LOGGER.info("Starting minion admin application on: {}", ListenerConfigUtil.toString(_listenerConfigs));
     _minionAdminApplication = new MinionAdminApiApplication(_instanceId, _config);
+    registerExtraComponents(_minionAdminApplication);
```
Comment (Jackie-Jiang) [inline pinot-minion/src/main/java/org/apache/pinot/minion/BaseMinionStarter.java:284]: Suggest modeling it as `createMinionAdminApp()` which returns `MinionAdminApiApplication` and can be overridden for flexibility. This way we can also support extending `MinionAdminApiApplication` if necessary. Same for other components.

Comment (Jackie-Jiang) [reply pinot-minion/src/main/java/org/apache/pinot/minion/BaseMinionStarter.java:284]: For broker, we can consider adding this new method as well, and mark `registerExtraComponents()` deprecated

Comment (soumitra-st) [reply pinot-minion/src/main/java/org/apache/pinot/minion/BaseMinionStarter.java]: Added in the latest commit.

File: pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java [hunk 2]
```diff
@@ -346,9 +346,7 @@ public void start()
       _sqlQueryExecutor = new SqlQueryExecutor(_spectatorHelixManager);
     }
     LOGGER.info("Starting broker admin application on: {}", ListenerConfigUtil.toString(_listenerConfigs));
-    _brokerAdminApplication =
-        new BrokerAdminApiApplication(_routingManager, _brokerRequestHandler, _brokerMetrics, _brokerConf,
-            _sqlQueryExecutor, _serverRoutingStatsManager, _accessControlFactory, _spectatorHelixManager);
+    _brokerAdminApplication = createBrokerAdminApp();
     registerExtraComponents(_brokerAdminApplication);
```
Comment (Jackie-Jiang) [inline pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java:350]: Let's move this call into default `createBrokerAdminApp()` as well

**Review by Jackie-Jiang — APPROVED** [review]
LGTM

File: pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java (hunk 2 above)
Comment (soumitra-st) [reply pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java:350]: Ah, typo.

Comment (soumitra-st) [reply pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java:350]: Done.

Comment (Jackie-Jiang) [reply pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java:350]: Have you pushed the change?

Comment (soumitra-st) [reply pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java]: Done now.

**Pull request merged**